engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugRenderingSystem.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.ingame.metrics;
+
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Sets;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.Share;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+
+@RegisterSystem
+@Share(DebugRenderingSystem.class)
+public class DebugRenderingSystem extends BaseComponentSystem {
+
+    private LinkedHashSet<MetricsMode> metricsModes;
+    private Iterator<MetricsMode> modeIterator;
+    private MetricsMode currentMode;
+
+    @Override
+    public void initialise() {
+        metricsModes = Sets.newLinkedHashSet();
+        modeIterator = Iterators.cycle(metricsModes);
+        currentMode = register(new NullMetricsMode());
+        register(new RunningMeansMode());
+        register(new SpikesMode());
+        register(new AllocationsMode());
+        register(new RunningThreadsMode());
+        register(new WorldRendererMode());
+        register(new NetworkStatsMode());
+        register(new RenderingExecTimeMeansMode("Rendering - Execution Time: Running Means - Sorted Alphabetically"));
+    }
+
+    public MetricsMode register(MetricsMode mode) {
```

Comment:
Ok, sounds like a separation of presentation/content issue. You'd like this class take care of the content (avilable through the MetricModes instances) and then DebugOverlay takes care of presenting the data on screen. 

Ok, I understand the approach and I'm fine with it. I would then recommend to rename the DebugRenderingSystem class to DebugMetricsSystem, so that it is not rendering-specific and is clearly associated with the concept of Debug Metrics.

Regarding the default metrics, I'd like to be clear that only the rendering-related metrics would be instantiated and registered during the rendering engine initialization. The DebugMetricsSystem by then should be up and running already. And on shutdown the rendering engine should make sure to unregister all its metrics. But this is out of the scope of this PR. I'm happy to leave those things as they are for the time being.

Finally, as this class is bound to stay, it will need some javadocs then. Class-level AND non-overriding public methods pleaaaaaaase!!! =)